images/virtualization-artifact/pkg/controller/service/vm_operation.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	virtv1 "kubevirt.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/common"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/powerstate"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+	"github.com/deckhouse/virtualization/api/core/v1alpha2/vmopcondition"
+)
+
+type VMOperationService struct {
+	client client.Client
+}
+
+func NewVMOperationService(client client.Client) VMOperationService {
+	return VMOperationService{
+		client: client,
+	}
+}
+
+func (s VMOperationService) getKVVM(ctx context.Context, vmNamespace, vmName string) (*virtv1.VirtualMachine, error) {
+	kvvm := virtv1.VirtualMachine{}
+	key := types.NamespacedName{Name: vmName, Namespace: vmNamespace}
+	err := s.client.Get(ctx, key, &kvvm)
+	if err != nil {
+		return nil, err
+	}
+	return &kvvm, nil
+}
```

Comment:
It seems to me that if the function can return not found error, then we should not check the object for nil.

So I see this method like this:
```go
kvvm := &virtv1.VirtualMachine{}
key := types.NamespacedName{Name: vmName, Namespace: vmNamespace}
err := s.client.Get(ctx, key, kvvm)
return kvvm, err
```
or
```
key := types.NamespacedName{Name: vmName, Namespace: vmNamespace}
return helper.FetchObject(ctx,
		key,
		client,
		&virtv1.VirtualMachine{})
```

Reply:
Changed to FetchObject